src/socket/socket.cr

```diff
@@ -33,16 +84,61 @@ lib C
   end
 
   fun htons(n : Int32) : Int16
-  fun bind(fd : Int32, addr : SockAddrIn*, addr_len : Int32) : Int32
+  fun bind(fd : Int32, addr : SockAddr*, addr_len : Int32) : Int32
   fun listen(fd : Int32, backlog : Int32) : Int32
-  fun accept(fd : Int32, addr : SockAddrIn*, addr_len : Int32*) : Int32
-  fun connect(fd : Int32, addr : SockAddrIn*, addr_len : Int32) : Int32
+  fun accept(fd : Int32, addr : SockAddr*, addr_len : Int32*) : Int32
+  fun connect(fd : Int32, addr : SockAddr*, addr_len : Int32) : Int32
   fun gethostbyname(name : UInt8*) : HostEnt*
+  fun getsockname(fd : Int32, addr : SockAddr*, addr_len : Int32*) : Int32
+  fun getpeername(fd : Int32, addr : SockAddr*, addr_len : Int32*) : Int32
   fun setsockopt(sock : Int32, level : Int32, opt : Int32, optval : Void*, optlen : Int32) : Int32
 
   SOCK_STREAM = 1
+  SOCK_DGRAM = 2
+  SOCK_RAW = 3
+
+  IPPROTO_IP = 0
+  IPPROTO_TCP = 6
+  IPPROTO_UDP = 17
+  IPPROTO_RAW = 255
+
   SOL_SOCKET = 0xffff
   SO_REUSEADDR = 0x0004
+
+  INET_ADDRSTRLEN = 16
+  INET6_ADDRSTRLEN = 46
+end
+
+class SocketError < Exception
+end
+
+class Socket < FileDescriptorIO
+  def afamily(family)
```

Comment:
It's ok. You can also do it like this:

```crystal
def afamily(family)
  C::AF_INET6.class.cast(family)
end
```

All Int and Float types have this class method to cast a value to their types (ir's written in Crystal so you can search for it, but as most things it's not yet documented, although it appears in the [API docs](http://crystal-lang.org/api/Int32.html) (but apparently there's a bug with the anchors)). In this case the class is known at compile-time (it's a constant) so the end result will be as efficient as a bitcast (no dispatch or method call involved when compiled in release mode).

You can make that method protected or private, I guess.